### ISSUE TITLE
TLCenter assumes balanced class priors: recentering on a class-imbalanced target buffer can reduce accuracy

### ISSUE THREAD

**Issue by Skiyoshika** [pyRiemann/pyRiemann#490]
### Summary

`TLCenter` recenters each domain to the identity using the **unweighted** Riemannian mean of that
domain's matrices. This implicitly assumes that the class priors are comparable across domains.
Under target **label shift** the estimated center is pulled toward the majority class, so the
recentring becomes a systematic rotation of the target distribution rather than a pure domain
correction.

The docstring doesn't currently mention this assumption (checked in 0.11 and in the 0.12 source).
It matters mostly in online / streaming use, where the "target domain" is a rolling buffer of
recent unlabelled trials whose class composition the system does not control.

The practical consequence has a fairly clean form: **the accuracy reached after centering depends
mainly on the buffer's class balance and is almost independent of the size of the domain shift.**
So whenever the un-centred baseline is already above that level, centering *hurts*.

I noticed this while using `TLCenter`-style recentring for online session adaptation, and it took
a while to work out why adaptation was sometimes worse than doing nothing — hence this issue,
mostly as a documentation request.

### Reproduction

Self-contained, uses `pyriemann.datasets.make_classification_transfer`; needs only pyriemann +
scikit-learn, no downloads. Full script at the end.

**A. The benefit erodes monotonically with buffer imbalance** (`domain_sep=3.0`, `TLCenter` fitted
on a 60-matrix unlabelled target buffer, 20 repetitions):

| class-1 fraction in buffer | target accuracy | vs no centering |
|---|---|---|
| 50% | 0.859 | +0.349 |
| 70% | 0.865 | +0.355 |
| 80% | 0.857 | +0.347 |
| 90% | 0.806 | +0.296 |
| 100% | 0.757 | +0.247 |

(no-centering baseline: 0.510)

**B. The sign flips when the domain shift is small:**

| `domain_sep` | no centering | balanced buffer | single-class buffer | Δ balanced | Δ single-class |
|---|---|---|---|---|---|
| 0.5 | 0.830 | 0.859 | 0.754 | +0.029 | **−0.076** |
| 1.0 | 0.790 | 0.859 | 0.757 | +0.069 | **−0.033** |
| 1.5 | 0.705 | 0.859 | 0.755 | +0.154 | +0.050 |
| 2.0 | 0.608 | 0.859 | 0.755 | +0.251 | +0.147 |
| 3.0 | 0.510 | 0.859 | 0.757 | +0.349 | +0.247 |

Note the structure: a **balanced** buffer lands at ~0.859 and a **single-class** buffer at ~0.755,
both essentially independent of `domain_sep`. Centering removes the domain shift either way; what
the imbalance adds is a roughly fixed penalty. Hence the crossover.

### Independent check on real data

Same effect in a motor-imagery setting (BCI IV-2a, cross-session T→E, binary, n=9, subject-level
paired t-tests), where the domain shift is small enough that the crossover condition is met. Only
the class composition of the 40-trial unlabelled adaptation buffer varies; evaluation is always on
the full balanced session:

| buffer composition | 50/50 | 70/30 | 80/20 | 90/10 | 100/0 |
|---|---|---|---|---|---|
| Δ vs no adaptation | +0.021 | −0.006 | −0.032 | −0.073 | **−0.115** |
| p | 0.21 | 0.61 | **0.045** | **0.004** | **0.002** |

Monotonic, and significant from 80/20 onward.

### Suggested resolution

**Minimal — documentation.** A note in `TLCenter` stating that the center is an unweighted mean
and therefore assumes comparable class priors across domains, and that under target label shift
the recentring is biased. Worth calling out for online use where buffer composition is
uncontrolled. I'd be happy to send a PR for just this.

**Optional — feature.** A class-prior-agnostic centering mode: estimate a per-class mean from
pseudo-labels supplied by the current model and combine them with equal weight. In my setting that
recovered most of the loss (worst case −0.115 → −0.020, all degradations becoming
non-significant) at a cost of −0.008 on balanced buffers. It's a first-order fix only —
pseudo-label errors leave a residual bias. A more principled treatment is Chu, Li, Kawanabe &
Kobler, *Rectifying Geometric Misalignment: Online Source-Free Adaptation for Class-Imbalanced
EEG* ([arXiv:2608.05315](https://arxiv.org/abs/2608.05315)), which adds a manifold-constrained
bias parameter to the tangent-space mapping.

I saw #464 / #471 touching `TLCenter`'s behaviour and docs recently — happy to follow whatever
direction you prefer here, or to drop this if it's already on the roadmap.

### Environment

```
pyriemann 0.11 (0.12 source also checked for the docstring)
numpy 1.26.4, scikit-learn, Python 3.11, Windows
```

### Reproduction script

<details>
<summary><code>repro_tlcenter.py</code></summary>

```python
"""Self-contained reproduction: TLCenter is harmful when the target buffer is
class-imbalanced and the domain shift is small. Needs only pyriemann + sklearn."""
import numpy as np
from sklearn.linear_model import LogisticRegression
from pyriemann.datasets import make_classification_transfer
from pyriemann.tangentspace import TangentSpace
from pyriemann.transfer import TLCenter, encode_domains, decode_domains

SRC, TGT = "source_domain", "target_domain"
N_MATRICES, BUFFER, REPS = 200, 60, 20


def make_domains(domain_sep, seed=42):
    X, y_enc = make_classification_transfer(
        n_matrices=N_MATRICES, class_sep=2.0, class_disp=1.0,
        domain_sep=domain_sep, theta=np.pi / 6, random_state=seed)
    X, y, d = decode_domains(X, y_enc)
    s = d == SRC
    return X[s], y[s], X[~s], y[~s]


def subsample(X, y, frac_first, n, rng):
    """n target matrices whose class-1 proportion is `frac_first` (simulated label shift)."""
    c = np.unique(y)
    k1 = min(int(round(frac_first * n)), int((y == c[0]).sum()))
    i1 = rng.choice(np.where(y == c[0])[0], k1, replace=False)
    i2 = rng.choice(np.where(y == c[1])[0], min(n - k1, int((y == c[1]).sum())), replace=False)
    return X[np.concatenate([i1, i2])]


def source_model(Xs, ys):
    ts = TangentSpace().fit(Xs)
    return ts, LogisticRegression(max_iter=1000).fit(ts.transform(Xs), ys)


def accuracy(ts, clf, X, y):
    return float((clf.predict(ts.transform(X)) == y).mean())


def recenter(Xs, buffer, Xt_eval):
    """Recenter source and target to identity using ONLY the unlabelled target buffer."""
    X = np.concatenate([Xs, buffer])
    dom = np.array([SRC] * len(Xs) + [TGT] * len(buffer))
    _, enc = encode_domains(X, np.array(["0"] * len(X)), dom)
    tl = TLCenter(target_domain=TGT)
    Xc = tl.fit_transform(X, enc)
    return Xc[:len(Xs)], tl.transform(Xt_eval)


def run(domain_sep, frac):
    Xs, ys, Xt, yt = make_domains(domain_sep)
    ts, clf = source_model(Xs, ys)
    base = accuracy(ts, clf, Xt, yt)
    accs = []
    for rep in range(REPS):
        rng = np.random.RandomState(100 + rep)
        buf = subsample(Xt, yt, frac, BUFFER, rng)
        Xs_c, Xt_c = recenter(Xs, buf, Xt)
        ts2, clf2 = source_model(Xs_c, ys)
        accs.append(accuracy(ts2, clf2, Xt_c, yt))
    return base, float(np.mean(accs))


def main():
    print("A) benefit erodes with buffer imbalance (domain_sep = 3.0)\n")
    print(f"   {'class-1 fraction':>17} {'accuracy':>10} {'vs no centering':>17}")
    base = None
    for frac in (0.5, 0.7, 0.8, 0.9, 1.0):
        base, acc = run(3.0, frac)
        print(f"   {frac:>17.0%} {acc:>10.3f} {acc - base:>+17.3f}")
    print(f"   (no centering baseline: {base:.3f})\n")

    print("B) the sign flips when the domain shift is small\n")
    print(f"   {'domain_sep':>11} {'no centering':>13} {'balanced':>10} {'single-class':>13}"
          f" {'delta bal':>10} {'delta single':>13}")
    for ds in (0.5, 1.0, 1.5, 2.0, 3.0):
        base, a_bal = run(ds, 0.5)
        _, a_one = run(ds, 1.0)
        print(f"   {ds:>11.1f} {base:>13.3f} {a_bal:>10.3f} {a_one:>13.3f}"
              f" {a_bal - base:>+10.3f} {a_one - base:>+13.3f}")


if __name__ == "__main__":
    main()
```

</details>

**Comment by qbarthelemy** [comment]
Thanks @Skiyoshika for your precise and detailed feedback!

As you can see in the documentation
https://pyriemann.readthedocs.io/en/latest/generated/pyriemann.transfer.TLCenter.html#pyriemann.transfer.TLCenter.fit
`fit()` function has accepted `sample_weight` as input argument since version 0.6, 
actually computing a weighted mean for each domain.

Does using these weights solve your problem of imbalanced classes?
Could you share the results of your experiments after using the weights?
